debug/slides_source/debug.tex

```diff
@@ -0,0 +1,382 @@
+    condition, mse, input, target)
+\end{lstlisting}
+\end{frame}
+
+\begin{frame}[fragile]
+  \frametitle{Breakpoint Op: example}
+\lstset{language=Python,
+        commentstyle=\itshape\color{blue},
+        stringstyle=\color{violet},
+        basicstyle=\footnotesize,
+        xleftmargin=-1em
+        }
+\begin{lstlisting}
+# Compile the theano function
+fct = theano.function([input, target], mse)
+
+# Use the function
+print fct([10, 0], [10, 5]) # Will NOT activate the breakpoint
+print fct([0, 0], [10, 5]) # Will activate the breakpoint
+\end{lstlisting}
+
+\end{frame}
+
+\subsection{PrintOp}
+\begin{frame}[fragile]
+  \frametitle{PrintOp}
+Allow to print during execution
+\lstset{language=Python,
+        commentstyle=\itshape\color{blue},
+        stringstyle=\color{violet},
+        basicstyle=\footnotesize,
+        xleftmargin=-1em
+        }
+\begin{lstlisting}
+import theano
+x = theano.tensor.vector()
+o = theano.printing.Print(``a message'')(x)
+f = theano.function([x], o)
+d = f([3, 4])
+a message  __str__ = [ 3.  4.]
+\end{lstlisting}
+\end{frame}
+
+
+\subsection{Nan}
+\begin{frame}
+  \frametitle{Most Frequent NaN Causes}
+\begin{itemize}
+\item Hyper parameter (ex: learning rate)
+\item Initialization of weight vectors
+\item Numerical Stability
+\item Algorithm Related
+\end{itemize}
+Run in NanGuardMode, DebugMode, or MonitorMode
+\end{frame}
+
+\begin{frame}[fragile]
+  \frametitle{NanGuardMode}
+\lstset{language=Python,
+        commentstyle=\itshape\color{blue},
+        stringstyle=\color{violet},
+        basicstyle=\footnotesize,
+        xleftmargin=-1em
+        }
+Can check:
+\begin{itemize}
+\item Nan
+\item Inf
+\item Big value (greater than 1e10)
+\end{itemize}
+\end{frame}
+
+\begin{frame}[fragile]
+  \frametitle{NanGuardMode: example}
+\lstset{language=Python,
+        commentstyle=\itshape\color{blue},
+        stringstyle=\color{violet},
+        basicstyle=\footnotesize,
+        xleftmargin=-1em
+        }
+\begin{lstlisting}
+x = T.dmatrix()
+w = theano.shared(numpy.random.randn(5, 7))
+y = T.dot(x, w)
+mode=NanGuardMode(nan_is_error=True,
+                  inf_is_error=True,
+                  big_is_error=True)
+fun = theano.function(
+    [x], y, mode=mode)
+infa = numpy.tile(
+    (numpy.asarray(100.) ** 1000000), (3, 5))
+fun(infa)
+\end{lstlisting}
+\end{frame}
+
+\subsection{Test Value}
+\begin{frame}[fragile]
+  \frametitle{Test Value}
+Set a value to symbolic variable and use it when building the graph.
+
+ \vspace{1cm}
+
+\lstset{language=Python,
+        commentstyle=\itshape\color{blue},
+        stringstyle=\color{violet},
+        basicstyle=\footnotesize,
+        xleftmargin=-1em
+        }
+\begin{lstlisting}
+# Can also be 'off', 'ignore', 'raise', 'pdb'
+theano.config.compute_test_value = 'warn'
+# input which will be of shape (5, 10)
+x  = T.matrix('x')
+# provide Theano with a default test-value
+x.tag.test_value = numpy.random.rand(5, 10)
```

Comment:
I think there's something missing with this example; now we have a variables with a test value... what happens next?